extension/ackextension/ackextension.go

```diff
@@ -7,12 +7,13 @@ package ackextension // import "github.com/open-telemetry/opentelemetry-collecto
 // to check the status of given ack ids.
 type AckExtension interface {
 	// ProcessEvent marks the beginning of processing an event. It generates an ack ID for the associated partition ID.
-	// ACK IDs are only unique within a partition. Two partitions can have the same ACK IDs but they are generated for different events.
+	// ACK IDs are only unique within a partition. Two partitions can have the same ACK IDs, but they are generated for different events.
 	ProcessEvent(partitionID string) (ackID uint64)
 
 	// Ack acknowledges an event has been processed.
 	Ack(partitionID string, ackID uint64)
 
 	// QueryAcks checks the statuses of given ackIDs for a partition.
+	// ackIDs that are not generated from ProcessEvent or have been removed as a result of previous calls to QueryAcks will return false.
 	QueryAcks(partitionID string, ackIDs []uint64) map[uint64]bool
```

Comment:
Can you please add `context.Context` to all the calls, even if it's ignored now? Just to follow best practices. Can be done in future